server/consumer.go

```diff
@@ -737,14 +736,14 @@ func (o *consumer) unsubscribe(sub *subscription) {
 	o.client.processUnsub(sub.sid)
 }
 
-// We need to make sure we protect access to the sendq.
+// We need to make sure we protect access to the outq.
 // Do all advisory sends here.
 // Lock should be held on entry but will be released.
 func (o *consumer) sendAdvisory(subj string, msg []byte) {
-	sendq := o.sendq
+	outq := o.outq
 	o.mu.Unlock()
-	if sendq != nil {
-		sendq <- &jsPubMsg{subj, subj, _EMPTY_, nil, msg, nil, 0}
+	if outq != nil {
+		outq.send(&jsPubMsg{subj, subj, _EMPTY_, nil, msg, nil, 0, nil})
```

Comment:
Since this will never block, you don't have to release lock anymore (still need to check for `outq != nil` since you set this to nil when stopping).
I am sure if we were using this everywhere (and not releasing lock to "send") then it may remove some obscure bugs that are due to unlocking/locking where state may have changed.

Reply:
Good idea, will also not nil it out anymore and set at construction.